testing/tests/ui/concat.stderr

```diff
@@ -0,0 +1,79 @@
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       "~b }}"
+ --> tests/ui/concat.rs:4:21
+  |
+4 | #[template(source = r#"{{ a~b }}"#, ext = "txt")]
+  |                     ^^^^^^^^^^^^^^
+
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       " ~b }}"
+  --> tests/ui/concat.rs:11:21
+   |
+11 | #[template(source = r#"{{ a ~b }}"#, ext = "txt")]
+   |                     ^^^^^^^^^^^^^^^
+
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       "~ b }}"
+  --> tests/ui/concat.rs:18:21
+   |
+18 | #[template(source = r#"{{ a~ b }}"#, ext = "txt")]
+   |                     ^^^^^^^^^^^^^^^
+
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       "~b~}}"
+  --> tests/ui/concat.rs:25:21
+   |
+25 | #[template(source = r#"{{~a~b~}}"#, ext = "txt")]
+   |                     ^^^^^^^^^^^^^^
+
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       " ~b~}}"
+  --> tests/ui/concat.rs:32:21
+   |
+32 | #[template(source = r#"{{~a ~b~}}"#, ext = "txt")]
+   |                     ^^^^^^^^^^^^^^^
+
+error: the concat operator `~` must be surrounded by spaces
+ --> <source attribute>:1:4
+       "~ b~}}"
+  --> tests/ui/concat.rs:39:21
+   |
+39 | #[template(source = r#"{{~a~ b~}}"#, ext = "txt")]
+   |                     ^^^^^^^^^^^^^^^
+
+error: failed to parse template source
```

Comment:
That's what I was afraid of. Do you think it's possible to improve error messages for both cases? Like saying that the `~` operator expects two items or something along the line.

Reply:
I guess in 9 out of 10 cases `{{ a ~ }}` is a misspelling of `{{ a ~}}` instead of a missing right hand operator. Same for `{{ ~ a }}` → `{{~ a}}`.

I don't know how to programmatically tell both cases apart or how to add a better error message here. :-/

Reply:
Right, we don't have enough context for that. Then let's keep it as is (sorry, just saw you added a commit ><).